### PR TITLE
Docs: keep the HTTP API list updated

### PR DESCRIPTION
**Pull request by ppcano**
I noticed the list of HTTP APIs needed an update. Changes in this PR include:

- Sorting the list of HTTP APIs
- Adding missing HTTP APIs
- Removing 404 APIs from the list
- Update the menu titles of two APIs to match the existing style

### CODE REVIEW

File: docs/sources/developers/http_api/preferences.md [hunk 1]
```diff
@@ -13,7 +13,7 @@ labels:
   products:
     - enterprise
     - oss
-title: 'HTTP Preferences API '
+title: 'Preferences API '
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/preferences.md:16]: nit: 
```suggestion
title: 'Preferences API'
```

File: docs/sources/developers/http_api/snapshot.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ labels:
   products:
     - enterprise
     - oss
-title: 'HTTP Snapshot API '
+title: 'Snapshot API '
```
Comment (jdbaldry) [inline docs/sources/developers/http_api/snapshot.md:16]: nit: 

```suggestion
title: 'Snapshot API'
```

**Review by jdbaldry — APPROVED** [review]
Thanks!

**Review by irenerl24 — APPROVED** [review]
LGTM! Thank you!

**Pull request merged**